design/ux/Obscuro_Gateway.md

```diff
@@ -0,0 +1,264 @@
+# The Obscuro Gateway - Design
+
+The scope of this document is to design a hosted [Wallet Extension](wallet_extension.md) called the "Obscuro Gateway" (OG).
+
+The OG will be a superset of the WE functionality, so this document will only cover the additions.
+
+## High level overview
+
+The OG will be a [Confidential Web Service](https://medium.com/p/983a2a67fc08), running inside SGX.
+
+The current WE is designed to be used by a single user holding multiple addresses accross potentially multiple wallets. 
+
+The OG must support mutiple users, each with multiple addresses. It can be seen as offering a WE per user.
+
+The Obscuro node has no concept of "User". It only authenticates based on the "blockchain address". 
+It expects to be supplied with a signed viewing key per address, so that it can respond encrypted with that VK. 
+
+*Note that multiple addresses can share a VK.*
+
+The role of the current WE is to manage a list of authenticated viewing keys (AVK), which it uses behind the scenes to communicate with an Obscuro node. 
+The AVKs are stored on the local computer in a file.
+
+The diagram below depicts the setup once the OG is implemented.
+![Architecture diagram](resources/og_arch.png)
```

Comment:
Can this uml code block be hidden ?

Reply:
I don't know how.

Reply:
You can use the `<details>` thingy :)

<details><summary>CLICK ME</summary>
<p>

#### yes, even hidden code blocks!

```python
print("hello world!")
```

</p>
</details>

Reply:
it doesn't look good with that.
I'll just leave both versions

Reply:
That was an example I randomly copied. I'd do something like:
<details><summary>Diagram's UML</summary>

```plantuml
@startuml
'https://plantuml.com/deployment-diagram
cloud "Obscuro Nodes"
actor Alice
component "Alice's Computer"{
    agent "Alice's MetaMask"
    node "Alice's Wallet Extension"
    database "Alice's Viewing Keys"
}
...
```
</p>
</details>

Think that would make the doc neater to read, but up to you really.